### ISSUE TITLE
whl compiled, still tries to install hdfs and gcp packages while --config=nogcp --config=nohdfs parameters were passed for build

### ISSUE THREAD

**Issue by arun-sl** [tensorflow/tensorflow#39620]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Fedora release 32
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: NA
- TensorFlow installed from (source or binary): source
- TensorFlow version:  r2.2
- Python version: 3.8.1
- Installed using virtualenv? pip? conda?: pip
- Bazel version (if compiling from source): 2.0.0
- GCC/Compiler version (if compiling from source): 10.0.1
- CUDA/cuDNN version: NA
- GPU model and memory: NA

**Describe the problem**
Even though tensorflow build command used had --config=nogcp --config=nohdfs parameters, the whl build while doing a pip install was trying to install dependencies like h5py and other gcp related packages. 

**Provide the exact sequence of commands / steps that you executed before running into the problem**
Have used following comand:

```
bazel build -s --config=noaws --config=nogcp --config=nohdfs --config=nonccl --config=opt //tensorflow/tools/pip_package:build_pip_package
```
```
./bazel-bin/tensorflow/tools/pip_package/build_pip_package /home/riscv/tensorflow_pkg
```
```
pip3 install /home/riscv/tensorflow_pkg/tensorflow-2.2.0-cp38-cp38-linux_riscv64.whl
```

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

```
[riscv@fedora-riscv tensorflow]$ pip3 install /home/riscv/tensorflow_pkg/tensorflow-2.2.0-cp38-cp38-linux_riscv64.whl                                           Defaulting to user installation because normal site-packages is not writeable
Processing /home/riscv/tensorflow_pkg/tensorflow-2.2.0-cp38-cp38-linux_riscv64.whl
Requirement already satisfied: astunparse==1.6.3 in /home/riscv/.local/lib/python3.8/site-packages (from tensorflow==2.2.0) (1.6.3)
Requirement already satisfied: absl-py>=0.7.0 in /home/riscv/.local/lib/python3.8/site-packages (from tensorflow==2.2.0) (0.9.0)
Collecting grpcio>=1.8.6
  Using cached grpcio-1.29.0.tar.gz (19.6 MB)
Requirement already satisfied: keras-preprocessing>=1.1.0 in /home/riscv/.local/lib/python3.8/site-packages (from tensorflow==2.2.0) (1.1.0)
Collecting gast==0.3.3
  Using cached gast-0.3.3-py2.py3-none-any.whl (9.7 kB)
Collecting h5py<2.11.0,>=2.10.0
  Using cached h5py-2.10.0.tar.gz (301 kB)
Requirement already satisfied: scipy==1.4.1; python_version >= "3" in /usr/lib64/python3.8/site-packages (from tensorflow==2.2.0) (1.4.1)
Requirement already satisfied: wrapt>=1.11.1 in /home/riscv/.local/lib/python3.8/site-packages (from tensorflow==2.2.0) (1.12.1)
Requirement already satisfied: termcolor>=1.1.0 in /home/riscv/.local/lib/python3.8/site-packages (from tensorflow==2.2.0) (1.1.0)
Collecting tensorboard<2.3.0,>=2.2.0
  Using cached tensorboard-2.2.1-py3-none-any.whl (3.0 MB)
Collecting google-pasta>=0.1.8
  Using cached google_pasta-0.2.0-py3-none-any.whl (57 kB)
Requirement already satisfied: protobuf>=3.8.0 in /home/riscv/.local/lib/python3.8/site-packages (from tensorflow==2.2.0) (3.11.3)
Requirement already satisfied: opt-einsum>=2.3.2 in /home/riscv/.local/lib/python3.8/site-packages (from tensorflow==2.2.0) (3.2.1)
Requirement already satisfied: six>=1.12.0 in /home/riscv/.local/lib/python3.8/site-packages (from tensorflow==2.2.0) (1.14.0)
Requirement already satisfied: numpy<2.0,>=1.16.0 in /home/riscv/.local/lib/python3.8/site-packages (from tensorflow==2.2.0) (1.18.3)
Requirement already satisfied: wheel>=0.26; python_version >= "3" in /home/riscv/.local/lib/python3.8/site-packages (from tensorflow==2.2.0) (0.34.2)
Requirement already satisfied: tensorflow-estimator<2.3.0,>=2.2.0rc0 in /home/riscv/.local/lib/python3.8/site-packages (from tensorflow==2.2.0) (2.2.0)
Collecting werkzeug>=0.11.15
  Using cached Werkzeug-1.0.1-py2.py3-none-any.whl (298 kB)
Requirement already satisfied: markdown>=2.6.8 in /home/riscv/.local/lib/python3.8/site-packages (from tensorboard<2.3.0,>=2.2.0->tensorflow==2.2.0) (3.2.2)
Collecting google-auth<2,>=1.6.3
  Using cached google_auth-1.14.3-py2.py3-none-any.whl (89 kB)
Collecting google-auth-oauthlib<0.5,>=0.4.1
  Using cached google_auth_oauthlib-0.4.1-py2.py3-none-any.whl (18 kB)
Collecting tensorboard-plugin-wit>=1.6.0
  Using cached tensorboard_plugin_wit-1.6.0.post3-py3-none-any.whl (777 kB)
Requirement already satisfied: setuptools>=41.0.0 in /home/riscv/.local/lib/python3.8/site-packages (from tensorboard<2.3.0,>=2.2.0->tensorflow==2.2.0) (46.1.3)
Requirement already satisfied: requests<3,>=2.21.0 in /home/riscv/.local/lib/python3.8/site-packages (from tensorboard<2.3.0,>=2.2.0->tensorflow==2.2.0) (2.23.0)
Collecting pyasn1-modules>=0.2.1
  Using cached pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
Collecting rsa<4.1,>=3.1.4
  Using cached rsa-4.0-py2.py3-none-any.whl (38 kB)
Collecting cachetools<5.0,>=2.0.0
  Using cached cachetools-4.1.0-py3-none-any.whl (10 kB)
Collecting requests-oauthlib>=0.7.0
  Using cached requests_oauthlib-1.3.0-py2.py3-none-any.whl (23 kB)
Requirement already satisfied: chardet<4,>=3.0.2 in /home/riscv/.local/lib/python3.8/site-packages (from requests<3,>=2.21.0->tensorboard<2.3.0,>=2.2.0->tensorflow==2.2.0) (3.0.4)
Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /home/riscv/.local/lib/python3.8/site-packages (from requests<3,>=2.21.0->tensorboard<2.3.0,>=2.2.0->tensorflow==2.2.0) (1.25.9)
Requirement already satisfied: certifi>=2017.4.17 in /home/riscv/.local/lib/python3.8/site-packages (from requests<3,>=2.21.0->tensorboard<2.3.0,>=2.2.0->tensorflow==2.2.0) (2020.4.5.1)
Requirement already satisfied: idna<3,>=2.5 in /home/riscv/.local/lib/python3.8/site-packages (from requests<3,>=2.21.0->tensorboard<2.3.0,>=2.2.0->tensorflow==2.2.0) (2.9)
Collecting pyasn1<0.5.0,>=0.4.6
  Using cached pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
Collecting oauthlib>=3.0.0
  Using cached oauthlib-3.1.0-py2.py3-none-any.whl (147 kB)
Building wheels for collected packages: grpcio, h5py
....
```

PS: I am trying to get Tensorflow built for RISCV architecture here.

**Comment by vnvo2409** [comment]
`h5py` isn't related to hdfs nor gcp. It is a Python package to save trained model in the format .h5

**Comment by arun-sl** [comment]
Thanks for your reply.
But installing this package resulted in error like no libhdfs.so found.

**Comment by vnvo2409** [comment]
could you be more specific ?

**Comment by ymodak** [comment]
@arun-sl Are you able to build TF successfully?

**Comment by arun-sl** [comment]
Hello All,

For the issue I have raised, the logs were as follows:

 Loading library to get version: libhdf5.so
  error: libhdf5.so: cannot open shared object file: No such file or
directory
  ----------------------------------------
  ERROR: Failed building wheel for h5py

This happened while doing following:

pip3 install
/home/riscv/tensorflow_pkg/tensorflow-2.2.0-cp38-cp38-linux_riscv64.whl

If needed I shall attach full logs as well.

Yes Yasir I was able to build the whl package.
But final installation I am still facing issues wrt dependencies not met.
